src/server/req_signal.c

```diff
@@ -135,43 +139,17 @@ req_signaljob(struct batch_request *preq)
 		}
 	}
 
-	if (jt == IS_ARRAY_NO) {
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
 
 		/* just a regular job, pass it on down the line and be done */
 
-		req_signaljob2(preq, parent);
-		return;
-
-	} else if (jt == IS_ARRAY_Single) {
-
-		/* single subjob, if running can signal */
-
-		offset = subjob_index_to_offset(parent, get_index_from_jid(jid));
-		if (offset == -1) {
-			req_reject(PBSE_UNKJOBID, 0, preq);
-			return;
-		}
-		i = get_subjob_state(parent, offset);
-		if (i == -1) {
-			req_reject(PBSE_IVALREQ, 0, preq);
-			return;
-		} else if (i == JOB_STATE_RUNNING) {
-			pjob = find_job(jid);		/* get ptr to the subjob */
-			if (pjob) {
-				req_signaljob2(preq, pjob);
-			} else {
-				req_reject(PBSE_BADSTATE, 0, preq);
-				return;
-			}
-		} else {
-			req_reject(PBSE_BADSTATE, 0, preq);
-			return;
-		}
+		req_signaljob2(preq, pjob);
```

Comment:
If subjob is not in JOB_STATE_RUNNING the request used to be rejected. Now it seems we are accepting it. Are we changing the behavior here?

Reply:
req is rejected at line 122 above just after call to chk_job_request()